controller/deployer/main.go

```diff
@@ -179,5 +180,17 @@ func (c *context) createDeploymentEvent(e ct.DeploymentEvent) error {
 		e.Status = "running"
 	}
 	query := "INSERT INTO deployment_events (deployment_id, release_id, job_type, job_state, status, error) VALUES ($1, $2, $3, $4, $5, $6)"
-	return c.db.Exec(query, e.DeploymentID, e.ReleaseID, e.JobType, e.JobState, e.Status, e.Error)
+	return c.execWithRetries(query, e.DeploymentID, e.ReleaseID, e.JobType, e.JobState, e.Status, e.Error)
+}
+
+var execAttempts = attempt.Strategy{
+	Total: 10 * time.Second,
+	Delay: 100 * time.Millisecond,
+}
+
+// Retry db queries in case postgres has been deployed
+func (c *context) execWithRetries(query string, args ...interface{}) error {
+	return execAttempts.Run(func() error {
+		return c.db.Exec(query, args...)
```

Comment:
On my journey I have seen quite a few different errors coming back from the DB during the update, and I found it hard to collate an appropriate list of retry-able errors, so I just went for a blanket retry here as no errors are usually expected.

We could make a whitelist of errors which should not be retried here, but given we don't usually expect an error here, what would be in the list?